src/main/java/org/quantumbadger/redreader/http/okhttp/OKHTTPBackend.java

```diff
@@ -31,11 +35,26 @@
 public class OKHTTPBackend implements HTTPBackend {
 
 	private final OkHttpClient mClient;
+	private static HTTPBackend httpBackend;
+	public static boolean reset = false;
+
+	public static HTTPBackend getHttpBackend(boolean force) {
+		if(force || httpBackend == null) {
+			httpBackend = new OKHTTPBackend();
+			reset = false;
+		}
+		return httpBackend;
+	}
 
 	public OKHTTPBackend() {
 
 		mClient = new OkHttpClient();
 
+		if(BaseActivity.getTorStatus()) {
+			Proxy tor = new Proxy(Proxy.Type.HTTP, new InetSocketAddress("127.0.0.1", 8118));
```

Comment:
If you can get it working with a SOCKS proxy, that's the better way to go because SOCKS is included directly in the Tor daemon, while the HTTP proxy is a separate daemon that wraps tor.  I have had little luck with SOCKS proxies in Android HTTP APIs.